File: relay/storage/backing/interface.js

class StorageInterface {

    constructor(label) {
        this.label = label;
    }

    async initialize() {
    }

    async set(ns, key, value) {
        throw new Error("Not Implemented");
    }

    async get(ns, key) {
        /* If key not found should throw ReferenceError */
        throw new Error("Not Implemented");
    }

    async has(ns, key) {
        throw new Error("Not Implemented");
    }

    async remove(ns, key) {
        throw new Error("Not Implemented");
    }

    async keys(ns, regex) {
        throw new Error("Not Implemented");
    }

    async shutdown() {
    }
}

module.exports = StorageInterface;

File: relay/storage/backing/interface.py
class StorageInterface(object):
    """ ABC for storage subsystem. """

    def __init__(self, label):
        self.label = label

    async def initialize(self):
        pass

    async def set(self, ns, key, value):
        raise NotImplementedError("Required impl by subclass");

    async def get(self, ns, key):
        """ If key not found should throw ReferenceError """
        raise NotImplementedError("Required impl by subclass");

    async def has(self, ns, key):
        raise NotImplementedError("Not Implemented");
        raise NotImplementedError("Required impl by subclass");

    async def remove(self, ns, key):
        raise NotImplementedError("Required impl by subclass");

    async def keys(self, ns, regex):
        raise NotImplementedError("Required impl by subclass");

    async def shutdown(self):
        pass
